homeassistant/components/homekit/type_air_purifiers.py

```diff
@@ -0,0 +1,452 @@
+"""Class to hold all air purifier accessories."""
+
+import logging
+from typing import Any
+
+from pyhap.const import CATEGORY_AIR_PURIFIER
+from pyhap.service import Service
+from pyhap.util import callback as pyhap_callback
+
+from homeassistant.const import STATE_ON
+from homeassistant.core import (
+    Event,
+    EventStateChangedData,
+    HassJobType,
+    State,
+    callback,
+)
+from homeassistant.helpers.event import async_track_state_change_event
+
+from .accessories import TYPES
+from .const import (
+    CHAR_ACTIVE,
+    CHAR_AIR_QUALITY,
+    CHAR_CURRENT_AIR_PURIFIER_STATE,
+    CHAR_CURRENT_HUMIDITY,
+    CHAR_CURRENT_TEMPERATURE,
+    CHAR_FILTER_CHANGE_INDICATION,
+    CHAR_FILTER_LIFE_LEVEL,
+    CHAR_NAME,
+    CHAR_PM25_DENSITY,
+    CHAR_TARGET_AIR_PURIFIER_STATE,
+    CONF_LINKED_FILTER_CHANGE_INDICATION,
+    CONF_LINKED_FILTER_LIFE_LEVEL,
+    CONF_LINKED_HUMIDITY_SENSOR,
+    CONF_LINKED_PM25_SENSOR,
+    CONF_LINKED_TEMPERATURE_SENSOR,
+    SERV_AIR_PURIFIER,
+    SERV_AIR_QUALITY_SENSOR,
+    SERV_FILTER_MAINTENANCE,
+    SERV_HUMIDITY_SENSOR,
+    SERV_TEMPERATURE_SENSOR,
+    THRESHOLD_FILTER_CHANGE_NEEDED,
+)
+from .type_fans import ATTR_PRESET_MODE, CHAR_ROTATION_SPEED, Fan
+from .util import cleanup_name_for_homekit, density_to_air_quality
+
+_LOGGER = logging.getLogger(__name__)
+
+CURRENT_STATE_INACTIVE = 0
+CURRENT_STATE_IDLE = 1
+CURRENT_STATE_PURIFYING_AIR = 2
+TARGET_STATE_MANUAL = 0
+TARGET_STATE_AUTO = 1
+FILTER_CHANGE_FILTER = 1
+FILTER_OK = 0
+
+
+@TYPES.register("AirPurifier")
+class AirPurifier(Fan):
+    """Generate an AirPurifier accessory for an air purifier entity.
+
+    Currently supports, in addition to Fan properties:
+    temperature; humidity; PM2.5; auto mode.
+    """
+
+    def __init__(self, *args: Any) -> None:
+        """Initialize a new AirPurifier accessory object."""
+        super().__init__(*args, category=CATEGORY_AIR_PURIFIER)
+        if self.preset_modes is None:
+            self.auto_preset = None
+        else:
+            self.auto_preset = next(
+                filter(lambda x: ("" + x).lower() == "auto", self.preset_modes), None
+            )
```

Comment:
Can you improve the readability of this code?

Reply:
Done, let me know what you think of the changed version